README.md
# TerminalStartup
A small script I made to help me differenciate which Ubuntu distro I just booted up, and/or which window has which distro, etc, etc.

I made the script because I have a few different Ubuntu distros installed on my Windows box, and sometimes I would get confused which distro I am working in, and/or which distro I just booted up.

This script will display a bar at the top of the console and by default, the script will print out the distro's name and version number.


<<<<<<< HEAD
#SYNTAX:
startbar [-b "BGCOLOR"]/[-f "FGCOLOR"]/[-t "TITLE"]
=======
# HOW TO USE:
This script is meant to be used with those Linux installs on Windows. One of the variables the script looks for is
the $WSL_DISTRO_NAME variable, which those Linux installs has, but are missing in regular Linux distros.

The variable contains the Linux distro name and version number, which the script uses when displaying the distro name
in the start up bar. 
>>>>>>> 75839f13f1ea1b9d3d031833232f1574311d05b3


CHANGELOG
v0.1.5.1
  -	Added datetime module
	  -	working on added option to display current date

v0.1.5
  - Added another argument (-r / --random), which when used, will randomize the
	coloring for both background and text

v0.1.4.6
  -	Added "random" color option for both BGCOLOR and FGCOLOR.
	  -	if user chooses "random" instead of a specific color code, the script
		will pick a random number anywhere between 0 - 232

v0.1.4.5
  -	Added default colors for Debian distro

v0.1.4.4
  -	Added a couple of examples to the end of the colormap
  -	edited the help screen so that the version option is displayed last


v0.1.4.3
  -	Added '-cmap' option that will display the colormap
	  -	displays the color codes (available color options)

v0.1.4.2
  -	Added '-v' option to display version number


v0.1.4.1
  -	To avoid any "case sensitivity" issues, I tweaked TITLE so that it is all
	lowercase letters when the function is searching the string to see if it
	contains "kali" or "ubuntu" in the name.

v0.1.4
  -	Renamed startbar.py to "startbar"
  -	Modified the script's permission and made it an executable file

v0.1.3.2
  -	Tweaked the default coloring used for Kali distros


v0.1.3
  -	Changed default BG color to 196 if the distro is neither Ubuntu or Kali
	  -	If Kali, the default BG color will be 51 (cyan - Kali's color scheme)
	  -	if Ubuntu, the default BG color will be 202 (orange - Ubuntu's color scheme)
	  -	If the '-b' option is used, the BG color will be over written with the option
		used.

v0.1.2
  -	Changed default BG color to 202


TO DO:
[ ]  -	Do something for when there isn't a WSL_DISTRO_NAME value
		  -	Set a default title string or something


startup.py
#!/usr/bin/env python3

import os
import shutil
import argparse

_VERSION_ = "0.4"

#class clr:
black = "16m"
red = "196m"
green = "46m"
yellow = "190m"
blue = "21m"
magenta = "201m"
cyan = "51m"
white = "15m"
BLACK = "16m\033[01m"
RED = "196m\033[01m"
GREEN = "46m\033[01m"
YELLOW = "190m\033[01m"
BLUE = "21m\033[01m"
MAGENTA = "201m\033[01m"
CYAN = "51m\033[01m"
WHITE = "15m\1033[01m"

HOME = os.getenv("HOME")
if os.getenv("WSL_DISTRO_NAME") is not None:
	DISTRONAME = os.getenv("WSL_DISTRO_NAME")
else:
	DISTRONAME = None

COLS, LINES = shutil.get_terminal_size()
parser = argparse.ArgumentParser(description='Simple app that draws a banner at the top of the terminal screen during start up, and displays which distro was just booted')

parser.add_argument('-t', dest="title", action="store", default=None, help="Sets title")
parser.add_argument('-fg', dest="font", action="store", default="7", help="Sets font color")
parser.add_argument('-bg', dest="barcolor", action="store", default="1", help="Sets bar color")
parser.add_argument('-b', '--bold', dest="boldfont", action="store_true", default="False", help="Font Format: Bold")
parser.add_argument('-i', '--italic', dest="italicfont", action="store_true", default="False", help="Font Format: Italics")
parser.add_argument('-u', '--underline', dest="underlinefont", action="store_true", default="False", help="Font Format: Underline")
parser.add_argument('1', nargs='+', help="\033[31mRed\033[m\n\tgde")

args = parser.parse_args()

BGCOLOR = ("\033[4" + str(args.barcolor) + "m")
FGCOLOR = ("\033[3" + str(args.font) + "m")
if args.boldfont is True:
	FGCOLOR = ("\033[01m" + FGCOLOR)

if args.italicfont is True:
	FGCOLOR = ("\033[03m" + FGCOLOR)

if args.underlinefont is True:
	FGCOLOR = ("\033[04m" + FGCOLOR)


if args.title is not None:
	DISTRONAME = args.title


def drawbanner():
	if DISTRONAME is not None:
		print("\033c\033[m" + BGCOLOR + " " * COLS + "\r\033[" + str(int((COLS - len(DISTRONAME))/2)) + "C" + FGCOLOR + DISTRONAME + "\033[m\n")
	else:
		print("\033c\033[m" + BGCOLOR + " " * COLS + "\033[m\n")

drawbanner()

colormapb.py
#!/usr/bin/env python3

import sys, os

initcount = 0
version = '0.5'

def cmap():
	print('[00;01mBasic Colors: [m')
	initcount = 0
	count = 7
	while count >= 0:
		print('[m  ' + '[' + str(len(str(initcount))) + 'D' + str(initcount) + ':[48;5;' + str(initcount) + 'm  [m ',end='')
		initcount += 1
		count -= 1
	print('')
	count = 7
	while count >= 0:
		print('[m  ' + '[' + str(len(str(initcount))) + 'D' + str(initcount) + ':[48;5;' + str(initcount) + 'm  [m ',end='')
		initcount += 1
		count -= 1
	print('')
	print('')
	print('[00;01mAdvanced colors: [m')
	initcount = 16
	while initcount < 232:
		countb = 6
		while countb > 0:
			count = 6
			while count > 0:
				print('[m   ' + '[' + str(len(str(initcount))) + 'D' + str(initcount) + ':[48;5;' + str(initcount) + 'm   [m ',end='')
				initcount += 1
				count -= 1
			countb -= 1
			print('')
		print('')
	print('[00;01mShades of grey: [m')
	while initcount < 256:
		count = 6
		while count > 0:
			print('[m   ' + '[' + str(len(str(initcount))) + 'D' + str(initcount) + ':[48;5;' + str(initcount) + 'm   [m ',end='')
			initcount += 1
			count -= 1
		print('')


cmap()


startbar.py
#!/usr/bin/env python3

"""
	Python script for use with Window's Linux installs that displays
	a bar at the top of the terminal, displaying the distro's name.
"""


import shutil
import os
import argparse

_VERSION_ = '0.1.3'		#	Current version number

COLS, LINES = shutil.get_terminal_size()	#	Setting values for COLS and LINES

TITLE = os.getenv("WSL_DISTRO_NAME")		#	Setting value for TITLE to be the value of $WSL_DISTRO_NAME

if "Ubuntu" in TITLE:	#	if the value of WSL_DISTRO_NAME has the word "Ubuntu" in it,
	BGCOLOR = "202"		#	the dafault bg color will be 202 (orange).
elif "Kali" in TITLE:	#	If the value has the word "Kali" in it, the default bg color
	BGCOLOR = "51"		#	will be '51' (cyan).
else:					#	Otherwise, the default color will be
	BGCOLOR = "196"		#	'196', red


parser = argparse.ArgumentParser(description='Short sample app')


parser.add_argument('-t', dest="title", action="store", default=TITLE, help="Sets title")
parser.add_argument('-f', dest="font", action="store", default="15", help="Sets font color")
parser.add_argument('-b', dest="barcolor", action="store", default=BGCOLOR, help="Sets bar color")

args = parser.parse_args()

BGCOLOR = str(args.barcolor)

print("\033c\033[48;5;" + BGCOLOR + "m" + " " * COLS + "\r\033[" + str(int((COLS - len(args.title))/2)) + "C\033[38;5;" + args.font + "m" + args.title + "\033[m\n")
